avro/avro_test.go

```diff
@@ -0,0 +1,488 @@
+	TeamName        string   `avro:"team_name"`
+	Owner           string   `avro:"ownerOfTeam"`
+	Sport           string   `avro:"kind_of_sport"`
+	URI             string   `avro:"uri"`
+	Players         []string `avro:"players"`
+	PlayerOfTheYear string   `avro:"-" json:"player_of_the_year"`
+	HasChangedName  bool     `avro:"has_changed_name"`
+	NumberOfPlayers int32    `avro:"number_of_players"`
+}
+
+type testData2 struct {
+	Manager  string `avro:"manager"`
+	TeamName string `avro:"team_name"`
+	Owner    string `avro:"owner"`
+	Sport    string `avro:"sport"`
+	Goals    int64  `avro:"goals"`
+}
+
+type testData3 struct {
+	Manager         string   `avro:"manager"`
+	TeamName        string   `avro:"teamName"`
+	Owner           string   `avro:"owner"`
+	Sport           string   `avro:"sport"`
+	Mascot          string   `avro:"mascot"`
+	WinningYears    []string `avro:"winning_years"`
+	AreGood         bool     `avro:"are_good"`
+	NumberOfPlayers int32    `avro:"number_of_players"`
+}
+
+type testData4 struct {
+	Manager         string   `avro:"manager"`
+	TeamName        string   `avro:"teamName"`
+	Owner           string   `avro:"owner"`
+	Sport           string   `avro:"sport"`
+	Mascot          string   `avro:"-"`
+	WinningYears    []string `avro:"winning_years"`
+	AreGood         bool     `avro:"are_good"`
+	NumberOfPlayers int32    `avro:"number_of_players"`
+}
+
+type testOptionalData struct {
+	Manager         string `avro:"manager,omitempty"`
+	TeamName        string `avro:"team_name"`
+	Owner           string `avro:"owner"`
+	Sport           string `avro:"sport"`
+	AreGood         bool   `avro:"are_good"`
+	NumberOfPlayers int32  `avro:"number_of_players"`
+}
+
+type TestParent struct {
+	TransactionURL string    `avro:"transaction_url"`
+	Presenter      TestChild `avro:"presenter"`
+}
+
+type TestChild struct {
+	Email string `avro:"email"`
+}
+
+type FilingReceived struct {
+	ApplicationID string `avro:"application_id"`
+	ChannelID     string `avro:"channel_id"`
+	Items         []Item `avro:"items"`
+}
+
+type Item struct {
+	SubmissionID string `avro:"submission_id"`
+}
+
+func TestUnitMarshal(t *testing.T) {
+	Convey("Correctly marshal avro to byte array", t, func() {
+		cs := &AvroMarshaller{
+			Schema: testSchema1,
+		}
+
+		cd := &testData1{
+			Manager:         "Pardew, Alan",
+			TeamName:        "Crystal Palace FC",
+			Owner:           "Long, Martin",
+			Sport:           "Football",
+			URI:             "/football/crystalpalace/123456789",
+			PlayerOfTheYear: "Yannick Bolasie",
+			HasChangedName:  false,
+			Players:         []string{"Hugo Lloris", "Harry Kane"},
+			NumberOfPlayers: int32(11),
+		}
+
+		bufferBytes1, err1 := cs.Marshal(cd)
+		So(err1, ShouldBeNil)
+		So(bufferBytes1, ShouldNotBeNil)
+	})
+
+	Convey("Marshal should return an error unless given a pointer to a struct", t, func() {
+		cs := &AvroMarshaller{
+			Schema: testSchema1,
+		}
+
+		test := "string"
+		bufferBytes1b, err1b := cs.Marshal(test)
```

Comment:
Can the variable names be consistent across the tests please?

Reply:
done